src/Domain/Stack/LiipTransformerStackExecutor.php

```diff
@@ -0,0 +1,186 @@
+     */
+    public function addPostProcessor(string $name, PostProcessor $postProcessor): void
+    {
+        $this->postProcessors[$name] = $postProcessor;
+    }
+
+    /**
+     * Apply the stack to the image.
+     */
+    public function apply(string $stackName, ImageReference $sourceImageReference): ImageReference
+    {
+        $config = $this->filterConfiguration->get($stackName);
+
+        $config += [
+            'quality' => 100,
+            'animated' => false,
+        ];
+
+        if ($sourceImageReference instanceof ImageReferenceFile) {
+            $image = $this->imagine->open($sourceImageReference->getPath());
+        } else {
+            $image = $this->imagine->load($sourceImageReference->getContent());
+        }
+
+        $image = $this->applyFilters($image, $config);
+        $resultImageReference = $this->exportConfiguredImageBinary($sourceImageReference, $image, $config);
+
+        return $this->applyPostProcessors($resultImageReference, $config);
+    }
+
+    public function applyFilters(ImageInterface $image, array $config): ImageInterface
+    {
+        foreach ($this->sanitizeFilters($config['filters'] ?? []) as $name => $options) {
+            $prior = $image;
+            $image = $this->filters[$name]->load($image, $options);
+
+            if ($prior !== $image) {
+                $this->destroyImage($prior);
+            }
+        }
+
+        return $image;
+    }
+
+    public function applyPostProcessors(ImageReference $image, array $config): ImageReference
+    {
+        foreach ($this->sanitizePostProcessors($config['post_processors'] ?? []) as $name => $options) {
+            $image = $this->postProcessors[$name]->process($image, $options);
+        }
+
+        return $image;
+    }
+
+    private function exportConfiguredImageBinary(ImageReference $imageReference, ImageInterface $image, array $config): ImageReference
+    {
+        // TODO: this is for now literal copy-paste from FilterManager
+        $options = [
+            'quality' => $config['quality'],
+        ];
+
+        if (\array_key_exists('jpeg_quality', $config)) {
+            $options['jpeg_quality'] = $config['jpeg_quality'];
+        }
+        if (\array_key_exists('png_compression_level', $config)) {
+            $options['png_compression_level'] = $config['png_compression_level'];
+        }
+        if (\array_key_exists('png_compression_filter', $config)) {
+            $options['png_compression_filter'] = $config['png_compression_filter'];
+        }
+
+        if ('gif' === $imageReference->getFormat() && $config['animated']) {
+            $options['animated'] = $config['animated'];
+        }
+
+        $filteredFormat = $config['format'] ?? $imageReference->getFormat();
+        try {
+            $filteredString = $image->get($filteredFormat, $options);
+        } catch (\Exception $exception) {
+            // TODO: why only a problem for webp but not png/jpg? and should the stack handle this
```

Comment:
we could also throw an exception (possibly containing the alternate gif image)

the problem with this is that we don't return the requested format. but from the comment below it seems we can't easily tell in advance if we do an animated gif...